frontend/rust-lib/flowy-database2/src/services/field/type_options/date_type_option/date_type_option.rs

```diff
@@ -306,16 +307,32 @@ impl TypeOptionCellDataCompare for DateTypeOption {
     &self,
     cell_data: &<Self as TypeOption>::CellData,
     other_cell_data: &<Self as TypeOption>::CellData,
+    sort_condition: SortCondition,
   ) -> Ordering {
     match (cell_data.timestamp, other_cell_data.timestamp) {
-      (Some(left), Some(right)) => left.cmp(&right),
+      (Some(left), Some(right)) => {
+        let order = left.cmp(&right);
+        match sort_condition {
+          SortCondition::Ascending => order,
+          SortCondition::Descending => order.reverse(),
+        }
+      },
       (Some(_), None) => Ordering::Greater,
       (None, Some(_)) => Ordering::Less,
       (None, None) => default_order(),
```

Comment:
There're many similar codes like this one.

```rs
match (a, b) => {
 some, some,
 some, none,
 none, some,
 none, none,
}
```

Can we use the built-in `cmp` function instead, and try to reduce the boilerplate code?

Reply:
In some cases, definitely. But most cases it's a bit difficult since the sort condition only applies to a single arm (some, some). What I can do is something like this

```rust
match (a, b) => {
  (some, some) => ...,
  _ => a.cmp(&b)
}
```

Or do you an alternative solution?

Reply:
I don't have a good solution for the case you mentioned. If that's the case, just leave it as is.

Reply:
Sounds good, resolving this thread for now!